oktapam/resource_database.go

```diff
@@ -0,0 +1,420 @@
+				Type:        schema.TypeList,
+				Required:    true,
+				MinItems:    1,
+				MaxItems:    1,
+				Description: descriptions.ManagementConnectionDetails,
+				Elem:        dbConnectionTypes,
+				ForceNew:    true,
+			},
+			attributes.ManagementGatewaySelector: {
+				Type:        schema.TypeMap,
+				Optional:    true,
+				Description: descriptions.ManagementGatewaySelector,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			attributes.ManagementGatewaySelectorID: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: descriptions.ManagementGatewaySelectorID,
+			},
+			attributes.CreatedAt: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: descriptions.CreatedAt,
+			},
+			attributes.UpdatedAt: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: descriptions.UpdatedAt,
+			},
+		},
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceDatabaseReadImport,
+		},
+	}
+}
+
+func resourceDatabaseCreate(ctx context.Context, d *schema.ResourceData, m any) diag.Diagnostics {
+	c := getSDKClientFromMetadata(m)
+	resourceGroupID := d.Get(attributes.ResourceGroup).(string)
+	projectID := d.Get(attributes.Project).(string)
+	canonicalName := d.Get(attributes.CanonicalName).(string)
+	dbType := d.Get(attributes.DatabaseType).(string)
+
+	var selectorLabels *map[string]string
+	if selector, ok := d.GetOk(attributes.ManagementGatewaySelector); ok {
+		if s, ok := selector.(map[string]any); ok {
+			tmpLabels := make(map[string]string, len(s))
+			for k, v := range s {
+				if value, ok := v.(string); ok {
+					tmpLabels[k] = value
+				}
+			}
+			selectorLabels = &tmpLabels
+		} else {
+			return diag.FromErr(fmt.Errorf("invalid label selector"))
+		}
+	}
+
+	mgmtType, mgmtDetails, err := mgmtConnectionDetailsFromResource(ctx, c, d)
+	if err != nil {
+		return err
+	}
+
+	if createdDb, err := client.CreateDatabase(ctx, c, resourceGroupID, projectID, canonicalName, dbType, mgmtType, *mgmtDetails, selectorLabels); err != nil {
+		return diag.FromErr(err)
+	} else if createdDb == nil {
+		d.SetId("")
+	} else {
+		d.SetId(createdDb.Id)
+	}
+
+	return resourceDatabaseRead(ctx, d, m)
+}
+
+func resourceDatabaseRead(ctx context.Context, d *schema.ResourceData, m any) diag.Diagnostics {
+	c := getSDKClientFromMetadata(m)
+	resourceGroupID := d.Get(attributes.ResourceGroup).(string)
+	projectID := d.Get(attributes.Project).(string)
+	databaseID := d.Get(attributes.ID).(string)
+
+	database, err := client.GetDatabase(ctx, c, resourceGroupID, projectID, databaseID)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(database.Id)
+
+	wrap := wrappers.DatabaseResourceResponseWrapper{*database}
+	attributeOverrides := utils.GenerateAttributeOverrides(d, wrap)
+	for key, value := range wrap.ToResourceMap(attributeOverrides) {
+		if err := d.Set(key, value); err != nil {
+			return diag.FromErr(err)
+		}
+	}
```

Comment:
This was done to make the setting of the password attribute more generic, and hopefully easier to plug in overrides for other resources when needed in the future.

Reply:
Could you explain what the deal was with password that required this?